Tasks/PullRequestCodeAnalysis/prcatask.ts

```diff
@@ -10,38 +12,52 @@ import { TaskLibLogger } from './TaskLibLogger';
 // Set up localization resource file
 tl.setResourcePath(path.join( __dirname, 'task.json'));
 
-var messageLimitInput:string = tl.getInput('messageLimit');
-var messageLimit: number = Number(messageLimitInput);
-if (isNaN(messageLimit)) {
-    // Looks like: "Expected message limit to be a number, but instead it was NOT_A_NUMBER"
-    tl.setResult(tl.TaskResult.Failed, tl.loc('Error_NotPullRequest', messageLimitInput));
-}
-
-var tlLogger: TaskLibLogger = new TaskLibLogger();
-var collectionUrl: string = tl.getVariable('System.TeamFoundationCollectionUri');
-var token: string = '';
-var repositoryId: string = tl.getVariable('Build.Repository.Id');
 
-var sourceBranch: string = tl.getVariable('build.sourceBranch');
+var sourceBranch: string = tl.getVariable('Build.SourceBranch');
 
-// Do not continue if the build was not triggered by a pull request
+// Do nothing (except log a message) if the build was not triggered by a pull request
 if (!sourceBranch.startsWith('refs/pull/')) {
     // Looks like: "Skipping pull request commenting - this build was not triggered by a pull request."
     console.log(tl.loc('Error_NotPullRequest'));
 } else {
-    var pullRequestId: number = Number(sourceBranch.replace('refs/pull/', ''));
+
+    var messageLimitInput:string = tl.getInput('messageLimit');
+    var messageLimit: number = Number(messageLimitInput);
     if (isNaN(messageLimit)) {
-        tl.debug(`Expected pull request ID to be a number. Actual: ${sourceBranch.replace('refs/pull/', '')}`);
+        // Looks like: "Expected message limit to be a number, but instead it was NOT_A_NUMBER"
```

Comment:
You also need to protect against a negative numbers and non-integers. http://stackoverflow.com/questions/10834796/validate-that-a-string-is-a-positive-integer